### ISSUE TITLE
Getting container status in the integration test is fragile

### ISSUE THREAD

**Issue by praveenkumar** [minishift/minishift#2557]
- https://github.com/minishift/minishift/blob/master/test/integration/testsuite/minishift.go#L88-L106 

we use as of now to get the status of a container which is fragile because here we check with what image name used to run the container and it is possible to be used the image-id instead name and run the container. The best way is if you already know a partial name of the container which you are interested in you can even use `docker ps -f "name=<name>"` and it will return the name if running.

**Comment by agajdosi** [comment]
@praveenkumar Issue might not be in `docker ps` vs `docker ps -f name=<part-of-name>` itself, but might be in fact that docker command for some reason fails soon after the stop-start.

If I can ask you for something, with your PR please try to add a fmt.Println() here https://github.com/minishift/minishift/blob/c4d3e2535ee83eaa58b68936750448ef5f7b3def/test/integration/util/util.go#L251 to see what we get from `docker ps`, if that's empty, please try to add minute or more sleep right after the stop->start and see if that passes: `And user waits for "60" seconds`.

Error capabilities of functions around the retry for status of containers are just poor/none. I will improve them as part of this issue, but tomorrow and on Friday there are bank holidays, so I could just give above steps ^

**Comment by amitkrout** [comment]
@praveenkumar  Curious to know why image name is changed after a restart, Is it the same container id origin-docker-registry before the restart?

**Comment by praveenkumar** [comment]
@amitkrout Image name doesn't change, it just origin container thought I will just use the Image id instead name and both are same things, but as I told the current implementation of getting container status is fragile and should be improved, I am sending PR soon for this.

**Comment by amitkrout** [comment]
Sure, please go ahead. I will have a look and understand the changes to clear my doubts.

**Comment by LalatenduMohanty** [comment]
@praveenkumar  you mean parsing through `docker ps` output is not a good  way to get status, right?

```
func (m *MinishiftRunner) GetOpenshiftContainers() string {
	cmdOut, _, _ := m.RunCommand("ssh -- docker ps")
	return strings.Trim(cmdOut, " \n")
}
```

**Comment by praveenkumar** [comment]
resolved by #2559
